components/common-go/baseserver/server.go

```diff
@@ -278,6 +282,12 @@ func (s *Server) initializeGRPC() error {
 			}),
 		),
 		grpcMetrics.UnaryServerInterceptor(),
+		grpc_recovery.UnaryServerInterceptor(grpc_recovery.WithRecoveryHandlerContext(
+			func(ctx context.Context, p interface{}) error {
+				logrus.Errorf("[PANIC] %s\n\n%s", p, string(debug.Stack()))
+				return status.Errorf(codes.Internal, "%s", p)
```

Comment:
I think grpc will already do this for you in a default middleware but I may be wrong. We could add a test which panics in the RPC handler and see